tests/test_basic.py

```diff
@@ -437,3 +440,37 @@ def needsUnittestUser(self):
         """Skips the test unless it is run as the ``unittest`` user."""
         if not testonly.is_unittest_user():
             self.skipTest("May fail with non-unittest users.")
+
+
+@pytest.fixture(scope="function")
+def with_metadata_checker(monkeypatch):
+    """Intercept requests to arrays and groups list endpoints and checks the URLs."""
+
+    def fake_and_check_request(self, method, url, *args, **kwargs):
+        assert "with_metadata=True" in url
```

Comment:
Here's the actual test assertion, inside the fixture. If it fails, pytest does the right thing and reports an error that's quite understandable.

Reply:
@ktsitsi @spencerseale @gspowley what do you think of this approach for testing our SDK - API interface? Is it one that you'd be comfortable using and maintaining with me?

Reply:
So basically, we mock the actual request and just ensure that whatever selection we're testing is passed into the request?

Reply:
@spencerseale correct. Actual integration tests are good, too, but these new tests run really fast and could help eliminate a lot of lurking generic bugs living between the high level Python API and the low level auto-generated API.

Reply:
I love the mocking! Great way to test and interate in the development and allow more comprehensive testing in the future without needing a full stack.

Reply:
Thanks @sgillies, I will be sure to follow this strategy.